src/headless/tasks/inasafe_wrapper.py

```diff
@@ -331,6 +352,11 @@ def generate_report(
     }
 
     """
+    # Clean up layer registry before using
+    # In case previous task exited prematurely before cleanup
+    layer_registry = QgsMapLayerRegistry.instance()
+    layer_registry.removeAllMapLayers()
+
```

Comment:
Cleanup before using

Reply:
because you are clearing layer registry here, could you add `add_impact_layers_to_canvas(impact_function)` below here https://github.com/inasafe/inasafe-headless/pull/30/files#diff-f36ef00976776f9d5600e763f4ac0b8cR370) (for single exposure) and move the loop here https://github.com/inasafe/inasafe-headless/pull/30/files#diff-f36ef00976776f9d5600e763f4ac0b8cR278 to https://github.com/inasafe/inasafe-headless/pull/30/files#diff-f36ef00976776f9d5600e763f4ac0b8cR373